al_aws.js

```diff
@@ -34,6 +36,132 @@ var selfUpdate = function (callback) {
 };
 
 
+var selfConfigUpdate = function (callback) {
+    async.waterfall([
+        function(callback) {
+            getConfigChanges(function(err, config) {
+                callback(err, config)
+            });
+        },
+        function(newValues, callback) {
+            getCurrentConfig(function(err, currentConfig) {
+                callback(err, newValues, currentConfig)
+            });
+        },
+        function(newValues, currentConfig, callback) {
+            applyConfigChanges(newValues, currentConfig, function(err, newConfig) {
+                callback(err, newConfig, currentConfig)
+            })
+        },
+        function(newConfig, currentConfig, callback) {
+            if (isConfigDifferent(newConfig, currentConfig)) {
+                var lambda = new AWS.Lambda();
+                var updateConfig = filterDisallowedConfigParams(newConfig);
+                return lambda.updateFunctionConfiguration(updateConfig, callback);
+            } 
+            else {
+                callback(null);
+            }
+        }
+    ],
+    function(err, config) {
+        if (err) {
+            console.info('Lambda self-update config error: ', err);
+        } else {
+            if (config !== undefined) {
+                console.info('Lambda self-update config successful. Config: ', config);
+            }
+            else {
+                console.info('Lambda self-update config nothing to update');
+            }
+        }
+        callback(err, config);
+    });
+};
+
+function getConfigChanges(callback) {
+    var s3 = new AWS.S3();
+    var configName;
+    
+    // Use this default config in order to update old collectors
+    if (process.env.aws_lambda_update_config_name == undefined) {
+        configName = DEFAULT_CONFIG_NAME;
+    }
+    else {
+        configName = process.env.aws_lambda_update_config_name 
+    }
+    
+    var params = {
+        Bucket: process.env.aws_lambda_s3_bucket,
+        Key: configName
+    };
+    s3.getObject(params, function(err, object) {
+        if (err) {
+            return callback(err);
+        } else {
+            let config = JSON.parse(object.Body.toString());
+            return callback(null, config);
+        }
+    });
+}
+
+function getCurrentConfig(callback) {
```

Comment:
Suggest, `getLambdaConfigLocal()`.

Reply:
I don't feel this is a better name for getting actual configuration. I am fine with `getActualConfig()` if you don't like `current`.